src/solc_0.8/defi/rules/ContributionRules.sol

```diff
@@ -0,0 +1,205 @@
+        IERC721 addrToMove = _listERC721Index[_listERC721Index.length - 1];
+        _listERC721Index[indexToDelete] = addrToMove;
+        _listERC721[addrToMove].index = indexToDelete;
+        _listERC721Index.pop();
+
+        emit ERC721MultiplierDeleted(address(reqContract));
+    }
+
+    function deleteMultiplierERC1155fromList(IERC1155 reqContract)
+        external
+        _isContract(address(reqContract))
+        onlyOwner
+    {
+        require(_islistERC1155Member(reqContract), "ContributionRules: contract is not in the list");
+        uint256 indexToDelete = _listERC1155[reqContract].index;
+        IERC1155 addrToMove = _listERC1155Index[_listERC1155Index.length - 1];
+        _listERC1155Index[indexToDelete] = addrToMove;
+        _listERC1155[addrToMove].index = indexToDelete;
+        _listERC1155Index.pop();
+
+        emit ERC1155MultiplierDeleted(address(reqContract));
+    }
+
+    function _islistERC721Member(IERC721 reqContract) internal view returns (bool) {
+        if (_listERC721Index.length == 0) return false;
+
+        return (_listERC721Index[_listERC721[reqContract].index] == reqContract);
+    }
+
+    function _islistERC1155Member(IERC1155 reqContract) internal view returns (bool) {
+        if (_listERC1155Index.length == 0) return false;
+
+        return (_listERC1155Index[_listERC1155[reqContract].index] == reqContract);
+    }
+
+    function _multiplierBalanceOfERC721(address account) internal view returns (uint256) {
+        uint256 _multiplier = 0;
+
+        for (uint256 i = 0; i < _listERC721Index.length; i++) {
+            IERC721 reqContract = _listERC721Index[i];
+
+            if (_listERC721[reqContract].balanceOf == true) {
+                _multiplier = _multiplier + _multiplierLogarithm(account, reqContract);
+            }
+
+            for (uint256 j = 0; j < _listERC721[reqContract].ids.length; j++) {
+                address owner = reqContract.ownerOf(_listERC721[reqContract].ids[j]);
+                if (owner == account) {
+                    _multiplier = _multiplier + _listERC721[reqContract].multipliers[j];
+                }
+            }
+        }
+
+        return _multiplier;
+    }
+
+    function _multiplierBalanceOfERC1155(address account) internal view returns (uint256) {
+        uint256 _multiplier = 0;
+        for (uint256 i = 0; i < _listERC1155Index.length; i++) {
+            IERC1155 reqContract = _listERC1155Index[i];
+
+            for (uint256 j = 0; j < _listERC1155[reqContract].ids.length; j++) {
+                uint256 _bal = reqContract.balanceOf(account, _listERC1155[reqContract].ids[j]);
+
+                if (_bal > 0) {
+                    _multiplier = _multiplier + _listERC1155[reqContract].multipliers[j];
+                }
+            }
+        }
+
+        return _multiplier;
+    }
+
+    function _multiplierLogarithm(address account, IERC721 contractERC721) internal view returns (uint256) {
```

Comment:
why not public ?

Reply:
done -> https://github.com/thesandboxgame/sandbox-smart-contracts/commit/3eebb77ee852e2dd4fb5f5a788b3ed2ac64fb5bc